docs/source/getting-started/first-feature.rst

```diff
@@ -5,124 +5,178 @@ Your First Feature
+In the context of a messaging feature, these are essentially the authorization and privacy concerns listed above.
 
-The approach we'll take is: a user Bob can message another user Alice if Alice has added Bob as a friend.
-Remember that friendships are single-directional in our app.
-So Alice adding Bob as a friend means that she gives permission (or *authority*) for Bob to send her a message.
+For the authorization part, we take the following approach: a user Bob can message another user Alice exactly when Alice has added Bob as a friend.
+When Alice adds Bob as a friend, she gives permission or *authority* to Bob to send her a message.
+It is important to remember that friendships can go in a single direction in our app.
+This means its possible for Bob to message Alice without Alice being able to message him back!
 
-In DAML this workflow is represented as a new choice on the ``User`` contract.
+To implement this workflow, we add a new choice to the ``User`` template.
+Copy and paste the following code at the end of the ``User.daml`` file.
+(Take care to match the indentation of the ``AddFriend`` choice, as the DAML parser pays attention to it.)
 
 .. literalinclude:: code/daml/User.daml
   :language: daml
   :start-after: -- SENDMESSAGE_BEGIN
   :end-before: -- SENDMESSAGE_END
 
-Let's break this down.
-The choice is ``nonconsuming`` because sending a message should not affect the existence of the ``User`` contract.
-By convention, the choice returns the ``ContractId`` of the resulting ``Message`` contract (which we'll show next).
-The parameters to the choice are the sender (the party wishing to talk to the signatory of this ``User`` contract) and the message text.
-The ``controller`` clause suggests that it is the ``sender`` who can exercise the choice.
-Finally, the body of the choice makes sure that the sender is a friend of the user and then creates the ``Message`` with the sender, receiver and content.
+As with the ``AddFriend`` choice, there are a few aspects to note here.
+
+    - The choice is ``nonconsuming`` because sending a message should not consume the ``User`` contract.
+    - By convention, the choice returns the ``ContractId`` of the resulting ``Message`` contract (which we'll show next).
+    - The parameters to the choice are the ``sender`` and ``content`` of this message.
+    - The ``controller`` clause suggests that it is the ``sender`` who can exercise the choice.
+    - The body of the choice first ensures that the sender is a friend of the user and then creates the ``Message`` contract with the ``receiver`` being the signatory of the ``User`` contract.
 
-Now let's see the ``Message`` contract template.
-This is very simple - data and no choices - as well as the ``signatory`` declaration.
+If you've added the ``SendMessage`` choice correctly, you should an error in the bottom *Problems* pane of Visual Studio Code that ``Message`` is undefined.
```

Comment:
```suggestion
If you've added the ``SendMessage`` choice correctly, you should see an error in the bottom *Problems* pane of Visual Studio Code that ``Message`` is undefined.
```

Reply:
Wouldn't it make sense to first add the `Message` template and then the `SendMessage` choice? Otherwise, we'll end up calling `create` for a template whose data model hasn't been defined yet.

Reply:
Yeah, that might be better.